src/workerd/server/server.c++

```diff
@@ -1304,79 +1304,72 @@ private:
   capnp::Capability::Client getCapability(uint channel) override {
     KJ_FAIL_REQUIRE("no capability channels");
   }
-  class CacheClientImpl final: public CacheClient {
+  class CacheClientImpl final : public CacheClient {
   public:
-    CacheClientImpl(Service& parent)
-        : parent(parent) {
-    }
+    CacheClientImpl(Service &cacheService) : cacheService(cacheService) {}
 
-    kj::Own<kj::HttpClient> getDefault(kj::Maybe<kj::String> cfBlobJson,
-                                       kj::Maybe<Tracer::Span&> parentSpan) override {
+    kj::Own<kj::HttpClient>
+    getDefault(kj::Maybe<kj::String> cfBlobJson,
+               kj::Maybe<Tracer::Span &> parentSpan) override {
 
       return kj::heap<CacheHttpClientImpl>(
-          parent, nullptr, kj::mv(cfBlobJson), kj::mv(parentSpan));
+          cacheService, nullptr, kj::mv(cfBlobJson), kj::mv(parentSpan));
     }
 
-    kj::Own<kj::HttpClient> getNamespace(
-        kj::StringPtr cacheName, kj::Maybe<kj::String> cfBlobJson,
-        kj::Maybe<Tracer::Span&> parentSpan) override {
-
-      auto encodedName = kj::encodeUriComponent(cacheName);
+    kj::Own<kj::HttpClient>
+    getNamespace(kj::StringPtr cacheName, kj::Maybe<kj::String> cfBlobJson,
+                 kj::Maybe<Tracer::Span &> parentSpan) override {
 
-      return kj::heap<CacheHttpClientImpl>(
-          parent,  kj::mv(encodedName), kj::mv(cfBlobJson), parentSpan);
+      return kj::heap<CacheHttpClientImpl>(cacheService, kj::str(cacheName),
```

Comment:
Oh, now I see why `cacheName` was being encoded here rather than later... because we have to allocate a copy anyway, so if we can combine the encoding step (which also allocates a copy) then we avoid a redundant copy. Sorry, I think I mislead you by suggesting that logic should have been moved later...